### ISSUE TITLE
Shares with "1/N" group specs can be created but are unrecoverable

### ISSUE THREAD

**Issue by ksmithbaylor** [BlockchainCommons/bc-sskr-rust#1]
Hello! This standard (and especially the Rust implementation) has proven to be very useful to me, and I hope it gains more adoption.

While working on a [simple CLI](https://github.com/ksmithbaylor/sskr-tool) that wraps this library for convenient use to split/recover a BIP-39 mnemonic, I noticed that if a `Spec` contains any `GroupSpec` with a member threshold of `1`, it will generate shares with no issues but then return a `ShamirError(ChecksumFailure)` when trying to combine shares. However, a `1/1` group seems to work fine. I checked this by trying to round-trip shares with every combination of `M` and `N`, and all were successful except where `M = 1` and `N > 1`.

The lack of support for `1/N` groups where `N > 1` could be inherent to the implementation of Shamir secret sharing, but if this is true, would it perhaps be a good idea to produce an error when trying to generate shares with a `Spec` containing such a group to prevent someone from accidentally creating unrecoverable shares? In my CLI, I [detect and reject](https://github.com/ksmithbaylor/sskr-tool/blob/1bc0d157cb926a862f3cc4d4d907586210c2ed35/src/split.rs#L75-L80) this case when splitting a mnemonic, but it would be easy for someone to unintentionally skip this check.

Code to reproduce:

```rust
use sskr::{sskr_combine, sskr_generate, Error, GroupSpec, Secret, Spec};
use std::str::from_utf8;

const TEXT: &str = "my secret belongs to me.";

fn roundtrip(m: usize, n: usize) -> Result<Secret, Error> {
    let secret = Secret::new(TEXT).unwrap();
    let spec = Spec::new(1, vec![GroupSpec::new(m, n).unwrap()]).unwrap();
    let shares: Vec<Vec<Vec<u8>>> = sskr_generate(&spec, &secret).unwrap();
    sskr_combine(&shares.iter().flatten().collect::<Vec<&Vec<u8>>>())
}

fn main() {
    // Good, uses a 2/3 group
    {
        let result = roundtrip(2, 3);
        assert_eq!(from_utf8(result.unwrap().data()).unwrap(), TEXT);
    }

    // Still ok, uses a 1/1 group
    {
        let result = roundtrip(1, 1);
        assert_eq!(from_utf8(result.unwrap().data()).unwrap(), TEXT);
    }

    // Bad, uses a 1/3 group
    {
        let result = roundtrip(1, 3);
        // ShamirError(ChecksumFailure)
        assert_eq!(format!("{}", result.err().unwrap()), "Checksum failure");
    }
}
```

**Comment by wolfmcnally** [comment]
@ksmithbaylor Thank you for reporting this! I just released 0.3.5, which includes the fix, and the unit test for your case is [here](https://github.com/BlockchainCommons/bc-sskr-rust/blob/83c0eb3100ae2815638b4a2944d3fe685122e3ff/src/lib.rs#L379).

**Comment by wolfmcnally** [comment]
@ksmithbaylor I should add that we do *not* recommend using or allowing 1-of-N splits, as every share is equivalent to the secret, which defeats the purpose of SSKR.
